lib/modules/manager/kustomize/artifacts.ts

```diff
@@ -0,0 +1,217 @@
+  if (!is.nonEmptyString(dependency.depName)) {
+    return;
+  }
+
+  if (dependency.depType !== 'HelmChart') {
+    return;
+  }
+
+  if (!is.nonEmptyArray(dependency.registryUrls)) {
+    return;
+  }
+
+  const currentChartExistingPath = await localExistingChartPath(
+    chartHome,
+    dependency.depName,
+    dependency.currentVersion,
+  );
+
+  if (!flagEnabled && is.nullOrUndefined(currentChartExistingPath)) {
+    logger.debug(
+      `Not inflating Helm chart for ${dependency.depName} as kustomizeInflateHelmCharts is not enabled and the current version isn't inflated`,
+    );
+    return;
+  }
+
+  if (
+    is.nonEmptyString(currentChartExistingPath) &&
+    is.nonEmptyString(dependency.newVersion)
+  ) {
+    logger.debug(`Deleting previous helm chart: ${currentChartExistingPath}`);
+    await deleteLocalFile(currentChartExistingPath);
+  }
+
+  const versionToPull = dependency.newVersion ?? dependency.currentVersion;
+
+  if (!is.nonEmptyString(versionToPull)) {
+    return;
+  }
+
+  const versionToPullExistingPath = await localExistingChartPath(
+    chartHome,
+    dependency.depName,
+    versionToPull,
+  );
+
+  if (is.nonEmptyString(versionToPullExistingPath)) {
+    logger.debug(
+      `Helm chart ${dependency.depName} version ${versionToPull} already exists at ${versionToPullExistingPath}`,
+    );
+    return;
+  }
+
+  const folderName = `${dependency.depName}-${versionToPull}`;
+  const untarDir = upath.join(chartHome, folderName);
+  const registryUrl = dependency.registryUrls[0];
+  logger.debug(
+    `Pulling helm chart ${dependency.depName} version ${versionToPull} to ${untarDir}`,
+  );
+  const cmd =
+    `helm pull --untar --untardir ${quote(untarDir)} ` +
+    `--version ${quote(versionToPull)} --repo ${quote(registryUrl)} ${dependency.depName}`;
+
+  await exec([cmd], execOptions);
+}
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  logger.debug(`kustomize.updateArtifacts(${packageFileName})`);
+  const project = parseKustomize(newPackageFileContent);
+  const isUpdateOptionInflateChartArchives =
+    config.postUpdateOptions?.includes('kustomizeInflateHelmCharts') === true;
+
+  if (is.nullOrUndefined(project)) {
+    return [
+      {
+        artifactError: {
+          stderr: 'Failed to parse new package file content',
+        },
+      },
+    ];
+  }
+
+  const chartHome = getSiblingFileName(
+    packageFileName,
+    project.helmGlobals.chartHome,
+  );
+
+  try {
+    const helmToolConstraint: ToolConstraint = {
+      toolName: 'helm',
+      constraint: config.constraints?.helm,
+    };
+
+    const execOptions: ExecOptions = {
+      docker: {},
+      userConfiguredEnv: config.env,
+      extraEnv: generateHelmEnvs(),
+      toolConstraints: [helmToolConstraint],
+    };
+
+    for (const dependency of updatedDeps) {
+      await inflateHelmChart(
```

Comment:
You're reading `project`, so use that to only call this function for updates which are chart references. return early if no chart was updated

Reply:
Am i understanding you correctly that i need to check that the `dependency` is referenced in the parsed kustomize `project`? Because there could be `HelmChart` deptypes present that aren't part of the kustomize project?

Reply:
there could be docker image updates and you should not try to use them as helm chart update